app/models/user.rb

```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```

Comment:
please put into file settings.yml. Use gem config to create that file(https://github.com/rubyconfig/config). Moreover, the order is constants, associations, validates, callbacks, enum, scope, ...
Remember the line break after each type

Reply:
done